README.md
# phaser-slot

history.txt
[20170717]

1. 사운드 재생 기능 추가
  - 바 당길때 게임 사운드 재생
  - 당첨 팝업 표시 될 때 사운드 추가

2. SuccessLine 표시 시간 줄이기
  - 팝업 표시 시간 당겨야함
  - 코인 나오는 시간 체크

3. 코인 떨어질때 사운드 재생 기능 추가


src/resources.js
var CONFIG = {
  _config: {
    'GAME_WIDTH': 720,
    'GAME_HEIGHT': 1230,
    'IMAGE_POSITION_X': 55,   //  첫번째 Reel 이미지의 X 좌표
    'IMAGE_SIZE_X': 217,      //  이미지의 가로 넓이
    'IMAGE_SIZE_Y': 220,      //  이미지의 세로 넓이
    'BAR_POSITION_Y': 704,    //  슬로 바의 Y좌표
    'BAR_POSITION_X': 273,
    'TOTAL_PLAY_COUNT': 3,      //  플레이 가능 횟수
    'DEFAULT_MUTE': false,  //  디폴트 사운드 켜기
    'REEL_POSITION_Y': 350,
    'COIN_TUNNEL_Y': 1025,
    'ITEM_TEXT_POSITION_X': 206,
    'HEADER_POSITION_X': 205,
    'HEADER_POSITION_Y': 435,
    'COIN_GROUP_POSITION_X': 110,
    'TALK_POSITION_X': 424,
    'TALK_POSITION_Y': 760,
    'SOUND_ONOFF_BUTTON_X': 570,
    'SOUND_ONOFF_BUTTON_Y': 15,
    'CLOSE_BUTTON_X': 640,
    'CLOSE_BUTTON_Y': 15,
    'ITEMS': ['heart','king','seven', 'queen', 'poker', 'poker', 'poker', 'poker']
    //'ITEMS': ['poker','poker','poker', 'poker', 'poker', 'poker', 'poker']
  },
  getConfig: function(key) {
    var configData = this._config[key];
    if(configData === undefined) {
      throw new Error('[' + key + '] is not included in the configuration!!');
    }
    return configData;
  }
};

var Message = {
  _message: {
    'modal.message.01': '축하합니다.\n{$}를 획득했습니다.',
    'modal.message.02': '한게임 포커에서 이어서 플레이',
    'modal.message.03': '아쉽지만\n모든 배팅 기회를\n사용 하셨습니다.'
  },
  getMessage: function(key, args) {
    var message = this._message[key];
    if(!message) {
      return '[ERROR] not defined "' + key + '"';
    }
    if(args && args.constructor === Array ) {
      for(var i = 0 ; i < args.length ; i++) {
        message = message.replace('{$}',args[i]);
      }
    }
    return message;
  }
};


src/Reel.class.js
function Reel(props) {
  this._imagePositionX = CONFIG.getConfig('IMAGE_POSITION_X');
  this._imageSizeX = CONFIG.getConfig('IMAGE_SIZE_X');
  this._imageSizeY = CONFIG.getConfig('IMAGE_SIZE_Y');
  this._slotPositionY = CONFIG.getConfig('REEL_POSITION_Y');
  this._firstReelSpeed = 90;
  this._secondReelSpeed = 150;
  this._thirdReelSpeed = 200;

  this.selectedImages = {};
  this.isComplete = false;

  this.game = props.game;
  items = props.items;

  //  first reel.
  this.reel1 = this.game.add.group();
  this.reel2 = this.game.add.group();
  this.reel3 = this.game.add.group();

  //  reel의 Image 생성
  this._createSlotImage(this._imagePositionX, items, this.reel1);
  this._createSlotImage(this._imagePositionX + this._imageSizeX, items, this.reel2);
  this._createSlotImage(this._imagePositionX + (this._imageSizeX * 2), items, this.reel3);
  this.reel1.changeable = true;
  this.reel2.changeable = true;
  this.reel3.changeable = true;

  this.blurX = game.add.filter('BlurX');
  this.blurY = game.add.filter('BlurY');
  this.reel1.filters = [this.blurX, this.blurY];
  this.reel2.filters = [this.blurX, this.blurY];
  this.reel3.filters = [this.blurX, this.blurY];
  this.blurX.blur = 0;
  this.blurY.blur = 0;

  this._initSlot();

}

Reel.prototype = {
  update: function() {
    if(this.isSpinning) {
      this._updateReel(this.reel1);
      this._updateReel(this.reel2);
      this._updateReel(this.reel3);
    }
  },

  startSpinning: function() {
    this._initSlot();
    this.isComplete = false;
    this.isSpinning = true;
    this.blurY.blur = 40;
  },

  stopSpinning: function() {
    this.blurY.blur = 0;
    this._slowReels();
  },

  getIsCompleteOnce: function() {
    var result = this.isComplete;
    this.isComplete = false;
    return result;
  },

  getSelected: function() {
    return this.selectedImages;
  },

  _updateReel: function(reel) {
    if(reel.changeable) {
      reel.changeable = false;
      var speed = reel.speed;
      var images = reel.children;
      var overImageInx = -1;

      for(var i = 0 ; i < images.length ; i++) {
        var image = images[i];
        image.y = image.y - speed;
        if(image.y <= -500) {
          overImageInx = i;
        }
      }

      if(overImageInx !== -1) {
        var idx = overImageInx - 1;
        if(overImageInx === 0) {
          idx = images.length - 1;
        }
        images[overImageInx].y = images[idx].y + image.height;
      }
      reel.changeable = true;
    }
  },

  _slowReels: function(){
    this.blurY.blur = 0;
    var me = this;
    var selectedImages = {
      'first': null,
      'second': null,
      'third': null,
    };

    var reel1stop = false;
    var reel2stop = false;
    var reel3stop = false;
    var slowReels = setInterval(function(){
      //  속도가 20이하로 느려지면 정확한 위치에 놓는다.
      if(me.reel1.speed <= 20) {
        reel1stop = true;
        me.reel1.speed = 0;
        if(!selectedImages.first) {
          selectedImages.first = me._getSelectedImage(me.reel1);
        }
      } else {
        me.reel1.speed -= 7;
      }

     if(me.reel2.speed <= 20) {
       reel2stop = true;
       me.reel2.speed = 0;
       if(!selectedImages.second) {
         selectedImages.second = me._getSelectedImage(me.reel2);
       }
     } else {
       me.reel2.speed -= 7
     }

     if(me.reel3.speed <= 20) {
       me.reel3.speed = 0;
       if(!selectedImages.third) {
         selectedImages.third = me._getSelectedImage(me.reel3, function(){
           reel3stop = true;
         });
       }
     } else {
       me.reel3.speed -= 7
     }

     if(reel1stop && reel2stop && reel3stop ) {
       clearInterval(slowReels);
       me.selectedImages = selectedImages;
       me.isSpinning = false;
       me.isComplete = true;
     }
   }, 100);
  },

  _initSlot: function() {
    //  슬롯 초기화
    this.reel1.speed = this._firstReelSpeed;
    this.reel2.speed = this._secondReelSpeed;
    this.reel3.speed = this._thirdReelSpeed;
    // align the reels vertically to their spots

    this.selectedImages = {};

    this.reel1.changeable = true;
    this.reel2.changeable = true;
    this.reel3.changeable = true;

    this.reel1.y = this._slotPositionY;
    this.reel2.y = this._slotPositionY;
    this.reel3.y = this._slotPositionY;
  },

  _shuffleRandom: function(n) {
    var returnArr = new Array();
    var temp;

    for(var i = 0 ; i < n ; i++){
      returnArr.push(i);
    }
    var randomN;
    for(var i = 0; i < returnArr.length ; i++) {
      randomN = Math.floor(Math.random() * n);
      temp = returnArr[i];
      returnArr[i] = returnArr[randomN];
      returnArr[randomN] = temp;
    }
    return returnArr;
  },

  _createSlotImage: function(x, items, reel) {
    var itemLength = items.length;
    var shuffleRandomArr = this._shuffleRandom(itemLength);
    for(var i = 0 ; i < itemLength ; i++){
      var name = items[shuffleRandomArr[i]];
      var slot = reel.create(x, (i * this._imageSizeY), name);
      slot.data = {'name': name};
    }
  },

  _getSelectedImage: function(reel, tweenOnComplete) {
    var moveY = 2000;
    var selectedIndex = 0;
    var reelImages = reel.children
    for(var i = 0 ; i < reelImages.length ; i++) {
      var imageY = reelImages[i].y;
      if(imageY > 0 && moveY > imageY) {
        moveY = imageY;
        selectedIndex = i;
      } else if(imageY === 0) {
        moveY = 0;
        break;
      }
    }
    var tween = this.game.add.tween(reel).to( { y: [reel.y - (this._imageSizeY + moveY)] }, 1000);
    tween.interpolation(function(v, k){
      return Phaser.Math.bezierInterpolation(v, k);
    });
    if(tweenOnComplete) {
      tween.onComplete.add(tweenOnComplete, this);
    }
    tween.start();

    //최종적으로 선택된 이미지
    if(selectedIndex === reelImages.length - 1) {
      selectedIndex = 0;
    } else {
      selectedIndex++;
    }
    return reelImages[selectedIndex].data.name;
  }
}


src/Modal.class.js
function Modal(props) {
  this.game = props.game;
  //this.modalName = props.modalName;
  this.modalName = '';

  this.isActivePresentModal = false;
  this.rays = {};
  this.mask = {};
  this.modal = {};
  this.modalCoin = {};

  this._modalPositionX = 80;
  this._modalPositionY = 230;

}

Modal.prototype = {
  update: function() {
    if(this.isActivePresentModal) {
      this.rays.angle++;
    }
  },

  showPresentModal: function(imageName) {
    this.isActivePresentModal = true;

    this.mask = game.add.graphics(0, 0);
    this.mask.beginFill(0x000000);
    this.mask.drawRect(0, 0, 720, 1350);
    this.mask.alpha = 0.7;

    this.modal = game.add.sprite(this._modalPositionX, this._modalPositionY, imageName + '_modal');
    this.modal.inputEnabled = true;
    this.modal.events.onInputDown.add(function() {
      window.open('http://play.google.com/store/apps/details?id=com.nhnent.Qpoker&hl=ko');
      //this.hideModal();
    }, this);

    this.rays = game.add.sprite(this._modalPositionX + 295, this._modalPositionY + 230, 'rays');
    this.rays.anchor.setTo(0.5, 0.5);
    this.modalCoin = this.game.add.sprite(this._modalPositionX + 200, this._modalPositionY + 180, 'modal_coin');
  },
  showGameOverModal: function() {
    this.mask = game.add.graphics(0, 0);
    this.mask.beginFill(0x000000);
    this.mask.drawRect(0, 0, 720, 1350);
    this.mask.alpha = 0.7;

    this.modal = game.add.sprite(this._modalPositionX, this._modalPositionY + 200, 'gameover-modal');
    this.modal.inputEnabled = true;
    this.modal.events.onInputDown.add(function() {
      window.open('http://play.google.com/store/apps/details?id=com.nhnent.Qpoker&hl=ko');

      //this.hideModal();
    }, this);
  },
  hideModal: function() {
    this.mask.destroy();
    this.modal.kill();
    if(this.modalCoin.kill) {
      this.modalCoin.kill();
    }
    if(this.rays.kill) {
      this.rays.kill();
    }
    this.isActivePresentModal = false;
  }
}


src/SlotBar.class.js
function SlotBar(props) {
  this._barPositionX = CONFIG.getConfig('BAR_POSITION_X');
  this._barPositionY = CONFIG.getConfig('BAR_POSITION_Y');
  this._imageSizeX = CONFIG.getConfig('IMAGE_SIZE_X');
  this._imageSizeY = CONFIG.getConfig('IMAGE_SIZE_Y');

  this.isActive = false;
  this.game = props.game;
  this.dragBarHandler = props.dragBarHandler;

  var bar = this.game.add.sprite(this._barPositionX, this._barPositionY, 'bar');
  this.stickUpImage = game.add.sprite(this._barPositionX + 62, this._barPositionY + 90, 'stick-up');
  this.stickDownImage = game.add.sprite(this._barPositionX + 62, this._barPositionY + 180, 'stick-down');
  this.stickDownImage.visible = false;

  this.barball = this.game.add.sprite(this._barPositionX + 31, this._barPositionY, 'barball');
  this.barball.inputEnabled = true;
  this.barball.anchor.set(0);
  this.barball.input.enableDrag(false, true, false, 0, null, bar);
  this.barball.input.allowHorizontalDrag = false;
  this.barball.events.onDragStart.add(this.onDragStart, this);
  this.barball.events.onDragStop.add(this.onDragStop, this);
  this.barball.events.onDragUpdate.add(this.onDragUpdate, this);
  this.barball.input.boundsRect = this.game.bounds;

  this.barball.events.onInputOver.add(function(){
    this.game.canvas.style.cursor = "pointer";
  }, this);
  this.barball.events.onInputOut.add(function(){
    this.game.canvas.style.cursor = "default";
  }, this);
}

SlotBar.prototype = {
  update: function() {
    if(this.isActive) {
      //  볼의 위치에 따라 스틱의 이미지를 교체한다.
      if(this.barball.y > this._barPositionY + 70) {
        this.stickUpImage.visible = false;
      } else {
        this.stickUpImage.visible = true;
      }

      if(this.barball.y > this._barPositionY + 200) {
        this.stickDownImage.visible = true;
      } else {
        this.stickDownImage.visible = false;
      }
    }
  },

  setDragEnable: function(flag) {
    this.barball.input.draggable = flag;
  },

  onDragStart: function(){
    playSound('lever-swipe');
    this.isActive = true;
  },

  onDragUpdate: function(me, point, x, y) {
    //this.game.debug.text("Distance to pointer: " + this.game.physics.arcade.distanceTo(ball), 32, 32);
  },

  onDragStop: function(){
    var tween = game.add.tween(this.barball).to( {x: this._barPositionX + 31, y: this._barPositionY }, 1000, Phaser.Easing.Bounce.Out, true);
    tween.onComplete.add(function() { this.isActive = false; }, this);
    this.dragBarHandler();
  }
}


src/ItemText.class.js
function ItemText(props) {
  var game = props.game;
  var textPositionX = CONFIG.getConfig('ITEM_TEXT_POSITION_X');
  var textPositionY = CONFIG.getConfig('COIN_TUNNEL_Y') + 83;

  var initPositionY = -50;
  this.itemCount = 4;
  this.items = new Array();
  for(var i = 0 ; i < this.itemCount ; i++) {
    var item = game.add.sprite(textPositionX, textPositionY + initPositionY, 'item_text_' + (i + 1).toString());
    initPositionY = initPositionY + 50;
    this.items.push(item);
  }

  game.time.events.loop(Phaser.Timer.SECOND * 3, function() {
    for(var i = 0 ; i < this.itemCount ; i++) {
      var item = this.items[i];
      var tween = game.add.tween(item).to( { y: item.y + 50 }, 1000, Phaser.Easing.Linear.None, true, 0);
      tween.onComplete.add(function() {
        if(this.y >= textPositionY + 150) {
          this.y = this.y - 200;
        }
      }, item);
    }
  }, this);
}

ItemText.prototype = {
  update: function() {
  },
  show: function() {
    for(var i = 0 ; i < this.itemCount ; i++) {
      this.items[i].alpha = 1.0;
    }
  },
  hide: function() {
    for(var i = 0 ; i < this.itemCount ; i++) {
      this.items[i].alpha = 0.0;
    }
  }
}


src/app.js
'use strict';
var reel = {};
var coinGroup = {};
var slotBar = {};
var successLine = {};
var reg = {};
var quake = {};
var talk = {};
var modal = {};
var itemText = {};

var slotBody = {};
var money = {};
var headerCoin = {};
var credit = {};
var bodyGroup = {};
var coinTunnel = {};

var gameWidthX = CONFIG.getConfig('GAME_WIDTH');
var gameHeightX = CONFIG.getConfig('GAME_HEIGHT');

var playCount = CONFIG.getConfig('TOTAL_PLAY_COUNT');
var isMute = CONFIG.getConfig('DEFAULT_MUTE');

var coinTunnelY = CONFIG.getConfig('COIN_TUNNEL_Y');
var reelPositionY = CONFIG.getConfig('REEL_POSITION_Y');

var coinGroupPositionX =  CONFIG.getConfig('COIN_GROUP_POSITION_X');
var reelFirstPostionX = CONFIG.getConfig('IMAGE_POSITION_X');

var headerPositionY = CONFIG.getConfig('HEADER_POSITION_X');
var headerPositionX = CONFIG.getConfig('HEADER_POSITION_Y');
var talkPositionX = CONFIG.getConfig('TALK_POSITION_X');
var talkPositionY = CONFIG.getConfig('TALK_POSITION_Y');

var soundOnOffButtonX = CONFIG.getConfig('SOUND_ONOFF_BUTTON_X');
var soundOnOffButtonY = CONFIG.getConfig('SOUND_ONOFF_BUTTON_Y');
var closeButtonX = CONFIG.getConfig('CLOSE_BUTTON_X');
var closeButtonY = CONFIG.getConfig('CLOSE_BUTTON_Y');

var isGameover = false;
var isGetProduct = false;

var game = new Phaser.Game(gameWidthX, gameHeightX, Phaser.CANVAS, '',
{
  preload: preload,
  create: create,
  update: update
}, false, false);

function preload() {
  game.load.image('background', 'assets/bg_select.png');
  game.load.image('slot-body', 'assets/SLOT_EMPTY_bg.png');
  //game.load.image('background', 'assets/bg.png');
  game.load.image('credit', 'assets/credit.png');
  game.load.image('money-15', 'assets/15000.png');
  game.load.image('money-10', 'assets/10000.png');
  game.load.image('money-5', 'assets/5000.png');
  game.load.image('money-0', 'assets/0.png');

  game.load.image('close-button', 'assets/close.png');
  game.load.image('sound-button-enable', 'assets/v_on.png');
  game.load.image('sound-button-disable', 'assets/v_off.png');

  game.load.image('poker', 'assets/item0.jpg');
  game.load.image('heart', 'assets/item1.jpg');
  game.load.image('king', 'assets/item2.jpg');
  game.load.image('seven', 'assets/item3.jpg');
  game.load.image('queen', 'assets/item4.jpg');
  game.load.image('success-line', 'assets/line.png');
  game.load.image('talk', 'assets/talk.png');
  game.load.image('item_text_1', 'assets/item1_text.png');
  game.load.image('item_text_2', 'assets/item2_text.png');
  game.load.image('item_text_3', 'assets/item3_text.png');
  game.load.image('item_text_4', 'assets/item4_text.png');

  game.load.image('rays', 'assets/rays.png');
  game.load.image('gameover-modal', 'assets/modal.png');
  game.load.image('modal_coin', 'assets/modal_coin.png');
  game.load.image('poker_modal', 'assets/item0_modal.png');
  game.load.image('heart_modal', 'assets/item1_modal.png');
  game.load.image('king_modal', 'assets/item2_modal.png');
  game.load.image('seven_modal', 'assets/item3_modal.png');
  game.load.image('queen_modal', 'assets/item4_modal.png');

  game.load.image('coin-tunnel', 'assets/coin_tunnel.png');
  game.load.image('coin', 'assets/small_coin.png');

  game.load.image('bar', 'assets/bar.png');
  game.load.image('barball', 'assets/barball.png');
  game.load.image('stick-up', 'assets/barstick_t.png');
  game.load.image('stick-down', 'assets/barstick_b.png');

  game.load.script('BlurX', 'src/BlurX.js');
  game.load.script('BlurY', 'src/BlurY.js');

  game.load.spritesheet('header-coin', 'assets/coin_ani.gif', 32, 31, 5);

  game.load.audio('speen-reel', 'assets/audio/scroll.mp3');
  game.load.audio('congratulation', 'assets/audio/ddadan.mp3');
  game.load.audio('coins', 'assets/audio/COIN.mp3');
  game.load.audio('lever-swipe', 'assets/audio/lever.mp3');
}

function create() {
  var margin = 50;
  var x = -margin;
  var y = -margin;
  var w = game.world.width + margin * 2;
  var h = game.world.height + margin * 2;
  game.world.setBounds(x, y, w, h);
  game.physics.setBoundsToWorld();
  //  Game에서 커서가 떠나도 게임이 정지하지 않는다.
  game.stage.disableVisibilityChange = true;

  var items = CONFIG.getConfig('ITEMS');

  reel = new Reel({
    'game': game,
    'items': items
  });
  game.add.sprite(0, 0, 'background');

  bodyGroup = game.add.group();

  slotBody = game.add.sprite(0, 0, 'slot-body');
  bodyGroup.add(slotBody);
  credit = game.add.sprite(headerPositionX + 190, headerPositionY, 'credit');
  credit.anchor.set(0.5, 0.5);
  bodyGroup.add(credit);
  money = game.add.sprite(headerPositionX, headerPositionY, 'money-' + (playCount * 5).toString());
  money.anchor.set(0.5, 0.5);
  bodyGroup.add(money);
  headerCoin = game.add.sprite(headerPositionX + 102, headerPositionY, 'header-coin');
  headerCoin.anchor.set(0.5, 0.5);
  bodyGroup.add(headerCoin);

  itemText = new ItemText({ game: game });
  coinTunnel = game.add.sprite(0, coinTunnelY, 'coin-tunnel');
  //mummy.scale.set(4);
  var anim = headerCoin.animations.add('walk');
  anim.play(5, true);

  coinGroup = game.add.group();
  coinGroup.enableBody = true;
  coinGroup.physicsBodyType = Phaser.Physics.ARCADE;

  slotBar = new SlotBar({
    'game': game,
    'dragBarHandler': spinReels
  });

  game.world.camera.position.set(0);

  successLine = game.add.sprite(reelFirstPostionX + 15, reelPositionY + 113, 'success-line');
  successLine.alpha = 0;

  modal = new Modal({game: game});

  talk = game.add.sprite(talkPositionX, talkPositionY, 'talk');
  var tween = game.add.tween(talk).to( { y:  talk.y - 10 }, 300, Phaser.Easing.Linear.None, false, 0, -1, true);
  tween.interpolation(function(v, k){
    return Phaser.Math.bezierInterpolation(v, k);
  });
  tween.start();

  game.sound.mute = isMute;
  var soundButtonImage = 'sound-button-enable';
  if(game.sound.mute) {
    soundButtonImage = 'sound-button-disable';
  }
  fsn.components.SoundOnOffButton({
      game: game,
      x: soundOnOffButtonX,
      y: soundOnOffButtonY,
      imageName: soundButtonImage
  });

  game.time.events.add(Phaser.Timer.SECOND * 5, showCloseButton, this);

  game.scale.scaleMode = Phaser.ScaleManager.SHOW_ALL;
  game.scale.pageAlignHorizontally = true;
  game.scale.pageAlignVertically = true;
}

function createCoins() {
  playSound('coins');
  for (var y = 0; y < 3; y++) {
    for (var x = 0; x < 15; x++) {
      var coin = coinGroup.create(coinGroupPositionX + x * 32, coinTunnelY + 62, 'coin');
      coin.alpha = 0;
      coin.name = 'alien' + x.toString() + y.toString();
      coin.checkWorldBounds = true;
      coin.events.onOutOfBounds.add(
        function (coin) {
          coin.alpha = 0;
          coin.reset(coin.x, coinTunnelY + 62);
          game.add.tween(coin).to( { alpha: 1 }, 300, Phaser.Easing.Linear.None, true, 0, 0);
          coin.body.velocity.y = 250 + Math.random() * 100;
        }, this);
      coin.body.velocity.y = 250 + Math.random() * 100;
      game.add.tween(coin).to( { alpha: 1 }, 300, Phaser.Easing.Linear.None, true, 0, 0);
    }
  }
}

function showCloseButton() {
  fsn.components.GameCloseButton({
      game: game,
      x: closeButtonX,
      y: closeButtonY,
      imageName: 'close-button'
  });
}

function update(){
  slotBar.update();
  reel.update();
  modal.update();

  //  결과를 한번만 가져온다.(연속적인 실행을 막기 위해)
  if(reel.getIsCompleteOnce()) {
    playCount--;
    slotBar.setDragEnable(true);
    var selectedImages = reel.getSelected();
    var isGetProduct = checkSlotPoint(selectedImages);

    //  당첨 되었을 경우
    if(isGetProduct) {
      this.game.time.events.add(Phaser.Timer.SECOND * 2, function(){
        playSound('congratulation');
      }, this);
      this.game.time.events.add(Phaser.Timer.SECOND * 2.5, function(){
        modal.showPresentModal(selectedImages.first);
        isGameover = true;
      }, this);
      slotBar.setDragEnable(false);
      return;
    }

    //  획득 상품도 없고 플레이 가능 횟수도 없을 경우 game over
    if(!isGetProduct && playCount <= 0) {
      this.game.time.events.add(Phaser.Timer.SECOND * 1, function(){
        modal.showGameOverModal();
      }, this);
      isGameover = true;
      slotBar.setDragEnable(false);
      return;
    }

    //대기 상태일 경우
    showSpritesForWaiting();
  }
}

function spinReels() {
  successLine.alpha = 0;

  hideSpritesForPlaying();

  money.loadTexture('money-' + ((playCount - 1) * 5).toString() , 0);
  playSound('speen-reel');

  addQuake();
  game.time.events.add(Phaser.Timer.SECOND * 0.6, function(){
    reel.stopSpinning();
  }, this);
  reel.startSpinning();
  slotBar.setDragEnable(false);
}

function checkSlotPoint(selectedImages) {
  var result = false;
  var firstName = selectedImages.first;
  var secondName = selectedImages.second;
  var thirdName = selectedImages.third;
  if(firstName === secondName && firstName === thirdName) {
    var tween = game.add.tween(successLine).to( { alpha: 1 }, 300, Phaser.Easing.Linear.None, false, 100, 3, false);
    tween.start();
    createCoins();
    result = true;
  }
  return result;
}

function showSpritesForWaiting() {
  talk.alpha = 1.0;
  itemText.show();
}

function hideSpritesForPlaying() {
  talk.alpha = 0.0;
  itemText.hide();
  coinGroup.killAll();
}

function addQuake() {
  var duration = 100;
  var ease = Phaser.Easing.Bounce.InOut;
  var autoStart = true;
  var delay = 0;
  var yoyo = true;
  var repeat = 12;

  quake = game.add.tween(bodyGroup).to({x: bodyGroup.x - 10}, duration, ease, autoStart, delay, repeat, yoyo);
}

function playSound(resouceName) {
  var music = game.add.audio(resouceName);
  music.play();
  //game.sound.play(resouceName);
  //game.sound.play(resouceName);
}
